TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/EditInduction/ExemptionReason.cshtml.cs

```diff
@@ -89,6 +146,54 @@ protected override async Task OnPageHandlerExecutingAsync(PageHandlerExecutingCo
             return;
         }
 
-        ExemptionReasons = await referenceDataCache.GetInductionExemptionReasonsAsync(activeOnly: true);
+        var exemptionReasons = await referenceDataCache.GetInductionExemptionReasonsAsync(activeOnly: true);
+
+        if (featureProvider.IsEnabled(FeatureNames.RoutesToProfessionalStatus))
+        {
+            RoutesWithInductionExemptions = DbContext.ProfessionalStatuses
+                .Include(p => p.RouteToProfessionalStatusType)
+                .ThenInclude(r => r!.InductionExemptionReason)
+                .Where(
+                    p => p.PersonId == PersonId &&
+                    p.RouteToProfessionalStatusType != null &&
+                    p.ExemptFromInduction == true &&
+                    p.RouteToProfessionalStatusType.InductionExemptionReason != null)
+                .Select(r => new RouteWithExemption()
+                {
+                    InductionExemptionReasonId = r.RouteToProfessionalStatusType!.InductionExemptionReasonId!.Value,
+                    RouteToProfessionalStatusId = r.RouteToProfessionalStatusTypeId,
+                    InductionExemptionReasonName = r.RouteToProfessionalStatusType.InductionExemptionReason!.Name,
+                    RouteToProfessionalStatusName = r.RouteToProfessionalStatusType.Name
+                });
+        }
+        if (RoutesWithInductionExemptions is not null && RoutesWithInductionExemptions.Any())
+        {
+            var exemptionReasonIdsToExclude = ExemptionReasonCategories.ExemptionsToBeExcludedIfRouteQualificationIsHeld
+                .Join(RoutesWithInductionExemptions,
+                    guid => guid,
+                    r => r.InductionExemptionReasonId,
+                    (guid, route) => route.InductionExemptionReasonId);
+
+            var exemptionReasonsToDisplay = ExemptionReasonCategories.ExemptionReasonIds
+                .Where(id => !exemptionReasonIdsToExclude.Contains(id))
+                .Join(exemptionReasons,
+                    guid => guid,
+                    exemption => exemption.InductionExemptionReasonId,
+                    (guid, exemption) => exemption)
+                .ToArray();
+
+            ExemptionReasons = ExemptionReasonCategories.CreateFilteredDictionaryFromIds(exemptionReasonsToDisplay);
+        }
+        else
+        {
+            var exemptionReasonsToDisplay = ExemptionReasonCategories.ExemptionReasonIds
```

Comment:
I've made this a front-end concern only but re-reading the ticket, it sounds like it's a business concern:
"As the QTLS and SET membership and Overseas Trained Teacher Induction exemption reasons can only apply at the route level we want to remove their editability from the person level field".
The ReferenceDataCache could have a different method for returning PersonLevelInduction reasons, which could be stored as part of the InductionExemptionReason data model ?

Reply:
Yep that sounds good to me.